src/vaultRegistry/types.ts

```diff
@@ -0,0 +1,70 @@
+import { ProtocolName, Token } from "src/types";
+import { Address } from "viem";
+
+export type TokenConstant = {
+  chains: number[];
+  address: { [key: string]: Address };
+  name: string;
+  symbol: string;
+  decimals: number;
+  logoURI: string;
+}
+
+export type VaultData = {
+  address: Address;
+  vault: Token;
+  asset: Token;
+  adapter: Token;
+  gauge?: Token;
```

Comment:
From a user perspective I'd expect these values to be addresses. That's the convention most other contracts/sdks follow.

I think it would be better to expose a function which takes an address as a parameter and returns `Token` for it, if it's an asset.